src/sentry/utils/services.py

```diff
@@ -84,6 +82,44 @@ def expose(self, context):
             else:
                 context[key] = getattr(base, key)
 
+    def replace(self, instance=None, backend_path=None, options=None):
```

Comment:
It might make more sense to expose this as a context manager so that you don't have to worry about people forgetting to restore it after they're done testing.

Reply:
The reason I didnt limit it to that is realistically this could be used outside of testing. I dont know why we'd want to do that (vs e.g. kicking a proc to change config), but its certainly plausible.

Reply:
Might be nice to have another method `replaced(instance, backend_path, options)` that can return a context manager.

```py
with my_service.replaced(mocked_service):
   # do whatever
```